### PR TITLE
docs: add external docs urls

### PR DESCRIPTION
**Pull request by mattrosno**
Contributes to https://github.com/carbon-design-system/carbon-platform/issues/1380

Adds `externalDocsUrl`s to the index, so the IBM Products library's assets have `View asset docs` in Carbon Platform like this example below.

#### What did you change?

- add external docs urls
- clean up ibm security carbon yml formatting
- fix smart quote in description

#### Example

image

### CODE REVIEW

**Comment by mattrosno** [comment]
@jeanservaas @aubrey-oneal @aagonzales can you review? Not all indexed assets are documented in https://pages.github.ibm.com/cdai-design/pal, but for those I found, I added as `externalDocsUrl`s.

File: packages/cloud-cognitive/carbon.yml [hunk 1]
```diff
@@ -186,6 +192,7 @@ assets:
       - input-control
   create-tearsheet-narrow:
     name: Create tearsheet narrow
+    externalDocsUrl: https://pages.github.ibm.com/cdai-design/pal/patterns/creation-flows/usage#wide-tearsheet
```
Comment (elycheea) [inline packages/cloud-cognitive/carbon.yml:195]:
```suggestion
    externalDocsUrl: https://pages.github.ibm.com/cdai-design/pal/patterns/creation-flows/usage#narrow-tearsheet
```

File: packages/cloud-cognitive/carbon.yml [hunk 2]
```diff
@@ -397,6 +416,7 @@ assets:
       - content-block
   page-header:
     name: Page header
+    externalDocsUrl: sign/pal/components/page-header/usage
```
Comment (elycheea) [inline packages/cloud-cognitive/carbon.yml:419]:
```suggestion
    externalDocsUrl: https://pages.github.ibm.com/cdai-design/pal/components/page-header/usage
```

**Review by elycheea — CHANGES_REQUESTED** [review]
Just two URLs to fix for Create tearsheet narrow and page header.

File: packages/cloud-cognitive/carbon.yml (hunk 2 above)
Comment (mattrosno) [reply packages/cloud-cognitive/carbon.yml:419]: Good catch - thanks!

**Review by elycheea — APPROVED** [review]

**Pull request merged**